src/tactic/abel.lean

```diff
@@ -189,8 +189,12 @@ meta def eval_neg (c : context) : normal_expr → tactic (normal_expr × expr)
   return (term' c (n', -n.2) x a',
     c.app ``term_neg c.inst [n.1, x, a, n', a', h₁, h₂])
 
-def smul {α} [add_comm_monoid α] (n : ℕ) (x : α) : α := n • x
-def smulg {α} [add_comm_group α] (n : ℤ) (x : α) : α := n • x
+@[nolint doc_blame] def nat_smul_inst {α} [add_comm_monoid α] : has_smul ℕ α := by apply_instance
```

Comment:
don't use `nolint doc_blame`, these should be in the nolints.json file (or documented).

Reply:
I'm confused, I thought `nolint doc_blame` was "this isn't deserving of documentation" whereas `nolints.json` is "I can't be bothered to document this.

These are not declarations that are intended as API; they only exist because it's easier/faster to construct terms with `def` than from within the meta code.

How do you feel about me just making them `private`? They're not something we want in the docs anyway, are they?

Reply:
Ah, `private` doesn't work with the name mangling. I've done what you suggest since it's consistent with how the rest of this file is handled.

I'd still argue that nolinting all of these helper declarations is the right thing to do, but not in this PR.

Reply:
> I'm confused, I thought `nolint doc_blame` was "this isn't deserving of documentation" whereas `nolints.json` is "I can't be bothered to document this.

I agree with this characterization. They are implementation details (a property they share with most declarations in this file) but the need for them is deserving of documentation.

Reply:
> but the need for them is deserving of documentation.

Indeed; probably as a library note. Maybe something like
```
-- see not [declarations for tactics] for an explanation of the lemmas and defs in this section
namespace _helpers

end _helpers
```
since I think the `_` prefix will rightly exclude them from doc-gen.